driver.go

```diff
@@ -273,40 +233,20 @@ func (d *DockerOnTop) Unmount(request *volume.UnmountRequest) error {
 	}
 	defer activemountsdir.Close() // There's nothing I could do about the error if it occurs
 
-	dirEntries, readDirErr := activemountsdir.ReadDir(2) // Check if there is any _other_ container using the volume
-	if len(dirEntries) == 1 || errors.Is(readDirErr, io.EOF) {
-		// If just one entry or directory is empty, unmount overlay and clean up
-
+	unmount, err := d.Deactivate(request, activemountsdir)
+	if err != nil {
+		return internalError("failed to deactivate the active mount:", err)
+	} else if unmount {
 		err = syscall.Unmount(d.mountpointdir(request.Name), 0)
 		if err != nil {
 			log.Errorf("Failed to unmount %s: %v", d.mountpointdir(request.Name), err)
 			return err
 		}
-
 		err = d.volumeTreePostUnmount(request.Name)
-		// Don't return yet. The above error will be returned later
-	} else if readDirErr == nil {
-		log.Debugf("Volume %s is still mounted in some other container. Indicating success without unmounting",
-			request.Name)
-	} else {
-		log.Errorf("Failed to list the activemounts directory: %v", err)
-		return internalError("failed to list activemounts/", err)
-	}
 
-	activemountFilePath := d.activemountsdir(request.Name) + request.ID
-	err2 := os.Remove(activemountFilePath)
-	if os.IsNotExist(err2) {
-		log.Warningf("Failed to remove %s because it does not exist (but it should...)", activemountFilePath)
-	} else if err2 != nil {
-		// Another pretty bad situation. Even though we are no longer using the volume, it is seemingly in use by us
-		// because we failed to remove the file corresponding to this container.
-		log.Criticalf("Failed to remove the active mount file: %v. The volume is now considered used by a container "+
-			"that no longer exists", err)
-		// The user most likely won't see this error message due to daemon not showing unmount errors to the
-		// `docker run` clients :((
-		return fmt.Errorf("docker-on-top internal error: failed to remove the active mount file: %w. The volume is "+
-			"now considered used by a container that no longer exists. Human interaction is required: remove the file "+
-			"manually to fix the problem", err)
+		log.Debugf("Unmounted volume %s", request.Name)
+	} else {
+		log.Debugf("Volume %s is still mounted. Indicating success without unmounting", request.Name)
```

Comment:
I don't agree with the error message change here. Here, you're seemingly saying "still mounted so I'm not unmounting it", which doesn't make sense. That's because of the ambiguity between "mounted" as in "a container is running where this volume is used" and "mounted" as in "the underlying overlay filesystem is present". IMO, the previous error message was clearer:
```go
log.Debugf("Volume %s is still mounted in some other container. Indicating success without unmounting", request.Name)
```

Maybe could be phrased even better, e.g., `Volume %s is still used in another container. Indicating success without unmounting`

Reply:
I like the one you proposed, it's more clear of what's happening:
197c6feb30581a079b201dab667119abb8009af1